### ISSUE TITLE
Warning: While building tokens.json, token collisions were found; output may be unexpected.

### ISSUE THREAD

**Issue by jsamr** [divriots/style-dictionary-to-figma#32]
I stumble on what seems to be a false positive warning. Let's say I have this style dictionary config:

```js
const { transform } = require("@divriots/style-dictionary-to-figma");

module.exports = {
  source: ["tokens/**/*.js"],
  format: {
    figmaTokensPlugin: ({ dictionary }) => {
      const transformedTokens = transform(dictionary.tokens, {
        cleanMeta: true,
      });
      return JSON.stringify(transformedTokens, null, 2);
    },
  },
  platforms: {
    css: {
      transformGroup: "css",
      buildPath: "css/",
      files: [
        {
          destination: "variables.css",
          format: "css/variables",
        },
      ],
      prefix: "qa",
    },
    figma: {
      buildPath: "figma/",
      files: [
        {
          destination: "tokens.json",
          format: "figmaTokensPlugin",
        },
      ],
    },
  },
};

```

In my tokens folder, I have:

```js
// radii.js
function radius(num) {
    return {
      value: `${num}px`,
      type: "borderRadius",
    };
  }
  
  module.exports = {
    radius: {
      xxsmall: radius(1),
      xsmall: radius(2),
      small: radius(4),
      medium: radius(8),
      large: radius(12),
      xlarge: radius(16),
      xxlarge: radius(32),
    },
  };
```

```js
// spacings.js
function spacing(num) {
  return {
    value: `${num}px`,
    type: "spacing",
  };
}

module.exports = {
  spacing: {
    xxxsmall: spacing(4),
    xxsmall: spacing(8),
    xsmall: spacing(12),
    small: spacing(16),
    medium: spacing(24),
    large: spacing(32),
    xlarge: spacing(40),
    xxlarge: spacing(48),
    xxxlarge: spacing(64),
  },
};
```

Running style dictionary:
```
> style-dictionary build --config style-dict.config.js


css
✔︎ css/variables.css

figma
⚠️ figma/tokens.json
While building tokens.json, token collisions were found; output may be unexpected.
    Output name xxsmall was generated by:
        radius.xxsmall   1px
        spacing.xxsmall   8px
    Output name xsmall was generated by:
        radius.xsmall   2px
        spacing.xsmall   12px
    Output name small was generated by:
        radius.small   4px
        spacing.small   16px
    Output name medium was generated by:
        radius.medium   8px
        spacing.medium   24px
    Output name large was generated by:
        radius.large   12px
        spacing.large   32px
    Output name xlarge was generated by:
        radius.xlarge   16px
        spacing.xlarge   40px
    Output name xxlarge was generated by:
        radius.xxlarge   32px
        spacing.xxlarge   48px
This many-to-one issue is usually caused by some combination of:
    * conflicting or similar paths/names in property definitions
    * platform transforms/transformGroups affecting names, especially when removing specificity
    * overly inclusive file filters
```

My understanding is that there should not be name collisions, since the paths are different (e.g. `radius.medium` vs `spacing.medium`). Moreover, CSS transform don't generate these warnings. Did I miss something, or is it a bug?

`style-dictionary@3.7.1` and `@divriots/style-dictionary-to-figma@0.4.0`

**Comment by jorenbroekema** [comment]
Hey, this actually also fails for the following platform configuration:

```js
{
  platforms: {
    figma: {
      buildPath: "figma/",
      files: [
        {
          destination: "tokens.json",
          format: "json",
        },
      ],
    },
  },
}
```
So the error you're getting is not related to this package.

I believe the issue is that you haven't configured a transform group for your figma "platform" object:
https://amzn.github.io/style-dictionary/#/transform_groups

In fact, you only need to add `transforms: ["name/cti/kebab"],` to your figma "platform" and things work okay, so I'm guessing there's an issue where the output names of the tokens collide, because `radius.medium` and `spacing.medium` both resolve to the same output name `medium`. This is not the case if you [include a transform like `name/cti/kebab`](https://amzn.github.io/style-dictionary/#/transforms?id=namectikebab) (or another one like name/cti/snake`), because that creates a name including a token's parent props, making it unique (`spacing-medium` and `radius-medium`). 

Hopefully this helps :)

**Comment by jsamr** [comment]
@jorenbroekema That's correct! I apologize for bothering, and thank you for your kind reply; I was missing transforms 🤦🏻

**Comment by jorenbroekema** [comment]
No worries, happy to help!
